pkg/convert/arb.go

```diff
@@ -0,0 +1,114 @@
+package convert
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"go.expect.digital/translate/pkg/model"
+)
+
+/* .arb files employ the key-value (key-translation) format, with separate files representing different languages.
+
+app_en.arb
+{
+	"title": "Hello World!",
+	"@title" : {
+		"description" : "Message to greet the World"
+	},
+	"greeting": "Welcome {user}!",
+	"@greeting": {
+		"placeholders": {
+			"user":{
+				"type":"string",
+				"example":"Bob"
+			}
+		}
+	}
+}
+
+app_fr.arb
+{
+  "title": "Bonjour le monde!",
+  "greeting": "Bienvenue {user}!"
+}
+
+*/
+
+// Converts a serialized data in ARB file format into model.Messages.
+func FromArb(data []byte) (model.Messages, error) {
+	var dst map[string]interface{}
+	if err := json.Unmarshal(data, &dst); err != nil {
+		return model.Messages{}, fmt.Errorf("unmarshal from ARB to model.Messages: %w", err)
+	}
+
+	findDescription := func(key string) (string, error) {
+		// if '@' prefix missing then no additional information is provided
+		subKey, ok := dst["@"+key]
+		if !ok {
+			return "", nil
+		}
+
+		keyMap, ok := subKey.(map[string]interface{})
+		if !ok {
+			return "", fmt.Errorf("unsupported value type '%T' for key '%s'", subKey, "@"+key)
+		}
+
+		description, ok := keyMap["description"]
+		if !ok {
+			return "", nil
+		}
+
+		if descriptionValue, ok := description.(string); ok {
+			return descriptionValue, nil
+		} else {
+			return "", fmt.Errorf("unsupported value type '%T' for key 'description'", descriptionValue)
+		}
+	}
+
+	var messages model.Messages
+
+	for key, value := range dst {
+		// Ignore a key if it begins with '@' as it only supplies additional information and not the actual message.
+		if key[0] == '@' {
+			continue
+		}
+
+		// If a key does not have an '@' prefix and its value is not of type string, then file is not formatted correctly.
+		value, ok := value.(string)
+		if !ok {
+			return model.Messages{}, fmt.Errorf("unsupported value type '%T' for key '%s'", value, key)
+		}
+
+		description, err := findDescription(key)
+		if err != nil {
+			return model.Messages{}, fmt.Errorf("find description of '%s': %w", key, err)
+		}
+
+		messages.Messages = append(messages.Messages, model.Message{
+			ID:          key,
+			Message:     value,
+			Description: description,
+		})
```

Comment:
The following reduces code boilerplate by reducing the number of variables.

```suggestion
		msg  := messages.Message{ID: key}
		
		msg.Message, ok = value.(string)
		if !ok {
			return model.Messages{}, fmt.Errorf("unsupported value type '%T' for key '%s'", value, key)
		}

		msg.Description, err = findDescription(key)
		if err != nil {
			return model.Messages{}, fmt.Errorf("find description of '%s': %w", key, err)
		}

		messages.Messages = append(messages.Messages, msg)
```